modules/apps/depot/depot-web/src/main/java/com/liferay/depot/web/internal/portlet/action/UpdateMembershipsMVCActionCommand.java

```diff
@@ -78,14 +90,23 @@ protected void doProcessAction(
 				birthdayCal.get(Calendar.DATE), birthdayCal.get(Calendar.YEAR),
 				contact.getSmsSn(), contact.getFacebookSn(),
 				contact.getJabberSn(), contact.getSkypeSn(),
-				contact.getTwitterSn(), user.getJobTitle(), groupIds,
-				user.getOrganizationIds(), null, null, user.getUserGroupIds(),
-				serviceContext);
+				contact.getTwitterSn(), user.getJobTitle(),
+				_getGroupIds(actionRequest, user), user.getOrganizationIds(),
+				null, null, user.getUserGroupIds(), serviceContext);
 		}
 		catch (NoSuchUserException | PrincipalException exception) {
 			SessionErrors.add(actionRequest, exception.getClass());
 
-			actionResponse.setRenderParameter("mvcPath", "/error.jsp");
+			if (exception instanceof PrincipalException.MustHavePermission) {
+				if (_log.isDebugEnabled()) {
+					_log.debug(exception);
+				}
+
+				sendRedirect(actionRequest, actionResponse);
+			}
+			else {
+				actionResponse.setRenderParameter("mvcPath", "/error.jsp");
+			}
```

Comment:
(suggestion) Doesn't it look better if we use a different catch clause? The `SessionErrors` chunk is duplicated, but I think the intent is more clear.
```
catch (PrincipalException.MustHavePermission principalException) {
	SessionErrors.add(actionRequest, exception.getClass());

	if (_log.isDebugEnabled()) {
		_log.debug(exception);
	}

	sendRedirect(actionRequest, actionResponse);
} catch (NoSuchUserException | PrincipalException exception) {
	SessionErrors.add(actionRequest, exception.getClass());

	actionResponse.setRenderParameter("mvcPath", "/error.jsp");
}
```

Reply:
I can change it, no hard thouths and I have to make other changes 🤷🏻‍♀️